xwiki-rendering-transformations/xwiki-rendering-transformation-macro/src/main/java/org/xwiki/rendering/internal/transformation/macro/MacroTransformation.java

```diff
@@ -388,7 +391,9 @@ public void transform(Block rootBlock, TransformationContext context) throws Tra
                     continue;
                 }
 
-                if (!((Macro<Object>) macro).isExecutionIsolated(macroParameters, macroBlock.getContent())) {
+                if (!((Macro<Object>) macro).isExecutionIsolated(macroParameters, macroBlock.getContent())
+                    && !this.isolatedExecutionConfiguration.isExecutionIsolated(macroBlock.getId()))
```

Comment:
To be exhaustive, we could add a unit test for this condition, the rest looks good. Thanks!

Reply:
Actually, after writing this, I found this limitation so bad that I changed it. It's now possible for the configuration to override the value provided by the macro in both ways (so the configuration can force a macro to be considered as not-isolated). The use case would be that a macro wrongly indicates that it is isolated, and the configuration would be a workaround until the macro has been fixed.